android/build.gradle

```diff
@@ -1,15 +1,26 @@
+import java.nio.file.Paths
+
+import org.apache.tools.ant.taskdefs.condition.Os
+import org.apache.tools.ant.filters.ReplaceTokens
+
+plugins {
+    id 'com.android.library'
+    id 'maven-publish'
+    id 'de.undercouch.download'
+    id 'com.diffplug.gradle.spotless' apply false
+}
+
+if (project == rootProject) {
+    apply from: 'spotless.gradle'
+}
```

Comment:
@ParaskP7 this plugin has been added in version `2.4.1` ([reference](https://github.com/software-mansion/react-native-reanimated/blob/e25d29aa689a1442d298927303473da44a07d475/android/build.gradle#L372-L374)), but it's only meant to be applied when building the library. Let me know if this is the proper way to apply it, thanks 🙇.

Reply:
👋 @fluiddot !

I am not sure why this `Spotless` plugin needs to be guarded with this `if (project == rootProject) { ... }` check, but we could maybe remove that check altogether and just have `Spotless` enabled by default. Then, this configuration will become much more straightforward. Wdyt? 🤔 

Do do so, we would need to:
1. Use `com.diffplug.spotless` instead of `com.diffplug.gradle.spotless` when defying the plugin version (`settings.gradle.kts`).
2. Use `id 'com.diffplug.spotless'` instead of `id 'com.diffplug.gradle.spotless' apply false` when applying the plugin (`build.gradle`).
3. Since the `spotless.gradle` is only used within the above `build.gradle` file, to move the `spotless { ... }` configuration directly into the `build.gradle` above and removing the `spotless.gradle` file altogether.

Reply:
Awesome @fluiddot , I double-checked that and can verify this change, thank you!

PS: Now we need to update the `aar` and `tarball` once more and we are done, right? 🤔

Reply:
> PS: Now we need to update the aar and tarball once more and we are done, right? 🤔

Yep, actually, I ran the `createNPMPackage` script in https://github.com/wordpress-mobile/react-native-reanimated/commit/734d44667145464f4006294b94644e929b53f28c to generate both AAR and tarball files but there were no modifications.

Reply:
Btw, I also updated the reference of this package in the Gutenberg PR ([reference](https://github.com/WordPress/gutenberg/pull/39097/commits/4e9762762ba54f879beaebb8bba3007b5eac75cb)).